src/Microsoft.DotNet.VersionTools/Dependencies/IDependencyInfoProvider.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.VersionTools.Dependencies
+{
+    public interface IDependencyInfoProvider : IDependencyUpdater
+    {
+        /// <summary>
+        /// Create a dependency info that this updater depends on. An updater implements this when
```

Comment:
I wonder if this is mixing responsibilities too much.  Should an "updater" be responsible for creating these dependency infos?

It seems to me that the Updater's responsibility should be:  "given a set of dependency infos, update the repo".  But here we are adding the responsibility of also creating those dependency infos.  It seems like that should be separate.

Reply:
Agreed, the provider idea was a lopsided attempt at cohesion, and it would be better to keep infos and updaters distinct. Changing.